config/routes.rb

```diff
@@ -256,6 +256,9 @@
   # directions
   get "/directions" => "directions#search"
 
+  # set home location
+  match "/set_home_loc" => "user#set_home_location", :via => [:get, :post]
```

Comment:
Whatever the decision is about the request type to use, this should only match what is needed and not the other one.